### PR TITLE
macOS support with clang

### PR DESCRIPTION
**Pull request by Dahlgren**

### CODE REVIEW

File: sqf-value/value.h [hunk 1]
```diff
@@ -31,6 +31,8 @@
 #include <variant>
 #include <initializer_list>
 
+#undef assert
```
Comment (Dahlgren) [inline sqf-value/value.h]: assert macro is included elsewhere and conflicts with tester assert function

Comment (X39) [reply sqf-value/value.h:34]: this is the wrong place for that then tho, should be at the tests themself
not in the value header

Comment (X39) [inline sqf-value/value.h]: Should be placed in main.cpp instead

this (may) cause issues with people including this in their extension and use assert afterwards themself

**Review by X39 — CHANGES_REQUESTED** [review]

File: sqf-value/value.h (hunk 1 above)
Comment (Dahlgren) [reply sqf-value/value.h]: Oops, wrong file yeah 🙈

**Review by X39 — APPROVED** [review]

**Pull request merged**